DUI3-DX/Converters/Autocad/Speckle.Converters.AutocadShared/ToHost/Geometry/MeshToHostConverter.cs

```diff
@@ -0,0 +1,117 @@
+using Speckle.Converters.Common;
+using Speckle.Converters.Common.Objects;
+using Speckle.Core.Models;
+using Objects.Utils;
+using System.Linq;
+using Speckle.Core.Logging;
+using System.Collections.Generic;
+
+namespace Speckle.Converters.Autocad.Geometry;
+
+[NameAndRankValue(nameof(SOG.Mesh), NameAndRankValueAttribute.SPECKLE_DEFAULT_RANK)]
+public class MeshToHostConverter : ISpeckleObjectToHostConversion, IRawConversion<SOG.Mesh, ADB.PolyFaceMesh>
+{
+  private readonly IRawConversion<SOG.Point, AG.Point3d> _pointConverter;
+  private readonly IConversionContextStack<Document, ADB.UnitsValue> _contextStack;
+
+  public MeshToHostConverter(
+    IRawConversion<SOG.Point, AG.Point3d> pointConverter,
+    IConversionContextStack<Document, ADB.UnitsValue> contextStack
+  )
+  {
+    _pointConverter = pointConverter;
+    _contextStack = contextStack;
+  }
+
+  public object Convert(Base target) => RawConvert((SOG.Mesh)target);
+
+  /// <remarks>Mesh conversion requires transaction since it's vertices needed to be added into database in advance..</remarks>
+  public ADB.PolyFaceMesh RawConvert(SOG.Mesh target)
+  {
+    target.TriangulateMesh(true);
+
+    // get vertex points
+    using AG.Point3dCollection vertices = new();
+    List<AG.Point3d> points = target.GetPoints().Select(o => _pointConverter.RawConvert(o)).ToList();
+    foreach (var point in points)
+    {
+      vertices.Add(point);
+    }
+
+    ADB.PolyFaceMesh mesh = new();
+
+    using (ADB.Transaction tr = _contextStack.Current.Document.TransactionManager.StartTransaction())
```

Comment:
Is this where this is being called from? Did I see a transaction manage class? or was that somewhere else?